UserDocs.md

```diff
@@ -17,242 +17,228 @@ Install T4:
 $ pip install git+https://github.com/quiltdata/t4.git#subdirectory=ocean
 ```
 
-## API reference
+## User guide
+### Creating a package
 
-> T4 expects S3 paths of the form `BUCKET_NAME/path/to/dir/data.csv`.
+To create a new in-memory package you initialize a new Package object and then call mutators on it until you have the Package you want. Note that Packages are immutable, so set, update, and delete all return new Package instances. For example:
 
-### Reading and writing data
+	import t4
+	# initialize a package
+	p = t4.Package()
 
-![](img/helium-api.png)
+	# add entries individually using `set`
+	p = p.set("foo.csv", "/path/to/local/disk/foo.csv")
+	p = p.set("bar.csv", "s3://bucket/path/to/cloud/bar.csv")
 
+	# add many entries at once using `update`
+	p = p.update({"baz.csv": "/path/to/baz", "bam.png": "/path/to/bam"})
 
-#### `helium.get(src [, snapshot=None | version=None])`
-Retrieves `src` object from T4 and loads it into memory. Returns a `(data, metadata)` tuple.
+	# or grab everything in a directory at once using capture
+	p = p.capture("/path/to/folder/with/stuff/")
 
-Does not work on all objects. For a list of supported objects, see [Serialization](#serialization).
+	# delete entries using `delete`
+	p = p.delete("bam.png")
 
-Pass a snapshot hash to the optional `snapshot` parameter to retrieve the state of an S3 object at a given snapshot. Pass a version hash to the optional `version` parameter to retrieve the state of an S3 object at a given version. Only one or none of these two parameters may be specified at a time.
 
-#### `helium.put(src, dest, meta={})`
-Writes the in-memory object `src` to the path `dest` in S3.
-Transparently serializes some object types.
-See [Serialization](#serialization) for details.
+### Managing a local package
+A local package is one which has not been published yet. Local packages contain local data, so they’re highly malleable and provide no inherent guarantees about the data they reference.
```

Comment:
Saying that local packages "provide no inherent guarantees" seems overly pessimistic. Are you referring to the fact that the underlying files can change?

Reply:
That is correct. The difference is between local files, which may change without warning; and remote files on T4, which are guaranteed to persist.

Reply:
Made this point clearer.